### DISCUSSION TITLE
WebDAV timestamps are shown in wrong TZ on Android

### DISCUSSION THREAD

**Discussion by Zimbelstern** [bitfireAT/dav4jvm#11] [Q&A] [answered]
Hello,

I have an issue with DAVx5 4.3.1-ose on LineageOS 20 (Android 13) and SabreDav 4.4.0 as a WebDAV server.
The last modified dates are off by two hours, which is the difference between my local time and UTC.

On the server:
`> stat -c '%Y' myfile`
`1683307068`

On the Android device
`> content query --uri content://at.bitfire.davdroid.webdav/document/15`
...`last_modified=1683299868000`...

I suspect this is an issue with DAVx5 because KDE Plasma's file manager Dolphin shows correct timestamps for files retrieved from WebDAV.

**Comment by exstntlmsnthrp** [comment]
Just another user trying to help. 

Unfortunately, I have no clue if it has anything to do with Davx5, but I suspect not likely.

The first thing I would check is the time on your server, especially if they're exactly two hours different and that's also the time difference between your location and UTC.

**Reply by Zimbelstern** [reply]
Thanks for trying to help. However, I can not find anything wrong with my server configuration. Everything else is working fine, my desktop WebDAV client saves correct timestamps.

In the meantime, I've dug a little deeper and contacted my server with the python library `webdav4`.
`client.ls("myfile", detail=True)` gives a correct datetime object with the UTC timestamp and `tzinfo=datetime.timezone.utc` I can not find anything suspicious with that.

That's why I suspect the DAVx5 client. And I've reproduced this on more the one device.

**Reply by exstntlmsnthrp** [reply]
Well, at least it's good to check. Hopefully the Davx5 developers will respond and be able to help you. 

I only suggested this because I have a server that does not like the locale settings I have when I ssh, so it is always resetting itself to UTC.

**Comment by rfc2822** [comment]
Hi,

Do you mean the _retrieved_ timestamps are wrong? Because DAVx⁵ doesn't store any timestamps on the server, this is the server's responsibility.

Can you please provide the `stat` and `content-query` like above plus `curl -vI -u user:pwd https://example.com/webdav/file` so that I can see the HTTP Last-Modified?

**Reply by Zimbelstern** [reply]
> Do you mean the _retrieved_ timestamps are wrong? Because DAVx⁵ doesn't store any timestamps on the server, this is the server's responsibility.

Exactly. I moved a new file via the AOSP file manager to the server at 19:26 local time (UTC+2). The file manager now shows 17:26 local time as last modified for this file in the WebDAV directory.

> Can you please provide the `stat` and `content-query` like above plus `curl -vI -u user:pwd https://example.com/webdav/file` so that I can see the HTTP Last-Modified?

```
> stat -c '%Y' myfile
1683825995
```

```
> content query --uri content://at.bitfire.davdroid.webdav/document/46
Row: 0 document_id=46, _display_name=myfile, mime_type=application/octet-stream, flags=454, _size=0, last_modified=1683818795000, icon=NULL, summary=NULL
```

```
> curl -vI -u user:pwd https://example.com/user/myfile
*   Trying 192.0.2.1:443...
* Connected to example.com (192.0.2.1) port 443 (#0)
* ALPN: offers h2,http/1.1
...TLS etc...
> HEAD /user/myfile HTTP/2
> Host: example.com
> authorization: Basic
> user-agent: curl/7.88.1
> accept: */*
> 
...TLS etc...
< HTTP/2 200 
HTTP/2 200 
< x-sabre-version: 4.4.0
x-sabre-version: 4.4.0
< last-modified: Thu, 11 May 2023 17:26:35 GMT
last-modified: Thu, 11 May 2023 17:26:35 GMT
< etag: "4da56923a8f054b0ab69c1a00cd446508901f5b2"
etag: "4da56923a8f054b0ab69c1a00cd446508901f5b2"
< content-type: application/octet-stream
content-type: application/octet-stream
< date: Thu, 11 May 2023 17:27:19 GMT
date: Thu, 11 May 2023 17:27:19 GMT
< server: lighttpd/1.4.59
server: lighttpd/1.4.59

< 
* Connection #0 to host example.com left intact

```
Thank you for looking into this.

**Reply by rfc2822** [reply, marked as answer]
Should have been fixed in #13 and should be in next DAVx5 version.
